code/modules/mob/living/simple_animal/abnormality/abno_cores.dm

```diff
@@ -0,0 +1,73 @@
+/obj/structure/abno_core
+	name = "blank abnormality core"
+	desc = "You shouldn't be seeing this. Please let someone know!"
+	icon = 'ModularTegustation/Teguicons/abno_cores/zayin.dmi'
+	icon_state = ""
+	anchored = FALSE
+	density = FALSE
+	resistance_flags = INDESTRUCTIBLE
+	pixel_x = -16
+	base_pixel_x = -16
+	var/mob/living/simple_animal/hostile/abnormality/contained_abno
+	var/release_time = 420 SECONDS//this is going to be reduced by a minute for every risk level
+	var/threat_level
+
+/obj/structure/abno_core/proc/Release()
+	if(!contained_abno)//Is this core properly generated?
+		return
+	new contained_abno(get_turf(src))
+	qdel(src)
+	return TRUE
+
+/obj/structure/abno_core/MouseDrop(over_object)
+	var/obj/machinery/abno_core_extractor/O = over_object//abno core extractor
+	if(istype(O) && Adjacent(O) && Adjacent(usr))
+		var/response = alert(usr,"Will you extract [src]?","This cannot be reversed.","Yes","No")
+		if(response == "Yes" && do_after(usr, 10, O))
+			Extract()
+			return
+	var/obj/structure/itemselling/I = over_object//item selling machine
+	if(istype(I) && Adjacent(I) && Adjacent(usr))
```

Comment:
Both of these require adjacent checks, these could be done earlier to save on repeat checks.
For example:
```
	if(!Adjacent(over_object) || !Adjacent(usr))
		return
```
This would just leave the type checks remaining.